### PR TITLE
Anagrams Counter Solution

### PR DESCRIPTION
**Pull request by maximorlenco**

### CODE REVIEW

File: MapsTasks/src/AnagramsCounter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return ret;
+    }
+
+    public static void populateMap(String[] input) {
+
+        for (String i : input) {
+            String word = transform(i);
+            if (!anagramsMap.containsKey(word)) {
+                Integer count = 0;
+                count++;
```
Comment (sergeylukichev) [inline MapsTasks/src/AnagramsCounter.java]: Why not initialise counter with 1 and drop count++ ?

File: MapsTasks/src/AnagramsCounter.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                anagramsMap.put(word, count);
+            } else {
+                Integer count = anagramsMap.get(word);
+                count++;
+                anagramsMap.put(word, count);
```
Comment (sergeylukichev) [inline MapsTasks/src/AnagramsCounter.java]: Overall, we have another notation in Java to express if statement: `if(a){t1} else {t2}` becomes `a?t1:t2`

You can replace everything above with just 2 lines of code.

**Comment by sergeylukichev** [comment]
👍

File: MapsTasks/src/AnagramsCounter.java (hunk 1 above)
Comment (maximorlenco) [reply MapsTasks/src/AnagramsCounter.java]: Oh, I somehow did not think

**Comment by maximorlenco** [comment]
https://github.com/maximorlenco/Java_Course/pull/21

вс, 3 февр. 2019 г. в 20:57, Sergey Lukichev <notifications@github.com>:

**Comment by maximorlenco** [comment]
@sergeylukichev can you check plz

File: MapsTasks/src/AnagramsCounter.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        String sorted = transform(s);
+        Integer ret = anagramsMap.get(sorted);
+        return ret;
+    }
+
+    public static void populateMap(String[] input,Map<String,Integer> anagramsMap) {
+
+            for (String i : input) {
+            String word = transform(i);
+                anagramsMap.put(word, !anagramsMap.containsKey(word) ? 1 : anagramsMap.get(word) + 1);
```
Comment (sergeylukichev) [inline MapsTasks/src/AnagramsCounter.java]: padding

**Comment by sergeylukichev** [comment]
As I understand, this supposes to be a solution for the words count: return how many times a name is in the original array?

The why do you:
1. Use Anagrams everywhere though the task is not about anagrams?
2. Why do you sort words when putting them to the map?

**Comment by maximorlenco** [comment]
@sergeylukichev  1.Yes, it was a task about the word counter, but I had to find them in an anagram array,
2.I sort them in order to be able to compare, since these are anagrams

**Comment by maximorlenco** [comment]
This is not a solution to the problem of JavaCollectionAssignments2019.This is from the last lesson when we improved the anagram task.But I will do this task too.

**Pull request merged**